packages/angular_devkit/benchmark/README.md

```diff
@@ -0,0 +1,113 @@
+
+It currently shows only time, process, cpu, and memory used but might be extended in the future.
+
+This tool was created to provide an objective and reproducible way of benchmarking process
+performance.
+
+Given a process (or its source code), process inputs and environment, keeping two of these elements
+constant while varying the third should allow for meaningful benchmarks over time.
+
+In the context of the DevKit, we publish many CLI tools and have access to their source code.
+By tracking tool resource usage we can catch performance regressions or improvements on our CI.
+
+
+## Installation
+
+You can install the benchmark tool via `npm install -g benchmark` for a global install, or without
+`-g` for a local install.
+Installing globally gives you access to the `benchmark` binary in your `PATH`.
+
+
+## CLI Usage
+
+Call the `benchmark` binary, followed by options, then double dash, then the command to benchmark:
+
+```
+$ benchmark -- node fibonacci.js 40
+[benchmark] Benchmarking process over 5 iterations, with up to 5 retries.
+[benchmark]   node fibonacci.js 40 (at D:\sandbox\latest-project)
+[benchmark] Process Stats
+[benchmark]   Elapsed Time: 2365.40 ms (2449.00, 2444.00, 2357.00, 2312.00, 2265.00)
+[benchmark]   Average Process usage: 1.00 process(es) (1.00, 1.00, 1.00, 1.00, 1.00)
+[benchmark]   Peak Process usage: 1.00 process(es) (1.00, 1.00, 1.00, 1.00, 1.00)
+[benchmark]   Average CPU usage: 4.72 % (5.03, 4.86, 4.50, 4.50, 4.69)
+[benchmark]   Peak CPU usage: 23.40 % (25.00, 23.40, 21.80, 21.80, 25.00)
+[benchmark]   Average Memory usage: 22.34 MB (22.32, 22.34, 22.34, 22.35, 22.35)
+[benchmark]   Peak Memory usage: 22.34 MB (22.32, 22.34, 22.34, 22.35, 22.35)
+```
+
+For more information on the available options, run `benchmark --help`:
+```
+$ benchmark --help
+[benchmark] benchmark [options] -- [command to benchmark]
+
+Collects process stats from running the command.
+
+Options:
+    --help                    Show this message.
+    --verbose                 Show more information while running.
+    --exit-code               Expected exit code for the command. Default is 0.
+    --iterations              Number of iterations to run the benchmark over. Default is 5.
+    --retries                 Number of times to retry when process fails. Default is 5.
+    --cwd                     Current working directory to run the process in.
+    --output-file             File to output benchmark log to.
+    --overwrite-output-file   If the output file should be overwritten rather than appended to.
+    --prefix                  Logging prefix.
+
+Example:
+    benchmark --iterations=3 -- node my-script.js
+```
+
+
+NB: the `fibonacci.js` in the example is a naive implementation of a fibonacci number calculator:
+```
+// fibonacci.js
+const fib = (n) => n > 1 ? fib(n - 1) + fib(n - 2) : n;
+console.log(fib(parseInt(process.argv[2])));
+```
+
+## API Usage
```

Comment:
I added this section after @hansl asked if it was possible to output in JSON. Using it like a library enables this case (and others).

I think what'll happens is that users that want to upload the results will use it as a library instead of calling the binary and parsing the file/stdout.

I think it's worthwhile to leave the usage listed in the readme as documentation for our own team. Then adding a warning/disclaimer below about stability and and keeping the package private should be enough for making our commitment explicit.